tests/generation/test_tf_utils.py

```diff
@@ -181,3 +183,152 @@ def test_validate_generation_inputs(self):
         with self.assertRaisesRegex(ValueError, "foo"):
             fake_model_kwargs = {"foo": "bar"}
             model.generate(input_ids, **fake_model_kwargs)
+
+    def test_transition_scores_greedy_search(self):
```

Comment:
For all tests added here: exact same tests and checks as in PT (including the numbers in the numerical checks) 🙏

Reply:
Nice!